app/app.go

```diff
@@ -680,17 +703,34 @@ func New(
 				versionMap := app.mm.GetVersionMap()
 
 				for moduleName := range versionMap {
-					if _, ok := fromVM[moduleName]; !ok {
-						fromVM[moduleName] = versionMap[moduleName]
-					}
+					fromVM[moduleName] = versionMap[moduleName]
 				}
+			} else {
+				println("Version map already initialized")
+
+				// Add defaults for staking subspace
+				stakingSubspace, _ := app.ParamsKeeper.GetSubspace(stakingtypes.ModuleName)
+				stakingSubspace.Set(ctx, stakingtypes.KeyMinCommissionRate, sdk.NewDec(0))
+
+				// Add defaults for resource subspace
+				resourceSubspace := app.GetSubspace(resourcetypes.ModuleName)
```

Comment:
Shouldn't we do the same for the did module?

Reply:
We can add it for safety.
Per our discussion, I was more worried on skipping InitGenesis for resource mostly.